### PR TITLE
DB-923 : 5.7 status variables are missing

### PR DESCRIPTION
**Pull request by george-lorch**
- 5.7 introduced a new field in the system status variables structure for scope (global|session|both). This field was not getting populated to a rational value in TokuDB yet since it didn't know about it.
- Fixed status reporting and set all scope values for status variables to SHOW_SCOPE_GLOBAL.

Tested locally, no mtr really applies.

### CODE REVIEW

**Comment by laurynas-biveinis** [comment]
Code change OK, but it is really suspicious that no MTR applies. In that case a testcase should be added, starting with 5.6. The list of status variables can be verified without their values, if that's more practical.

**Comment by george-lorch** [comment]
Well, I looked to InnoDB for an example to try to create a case and there is no similar case that I could find. I _believe_ the reasoning being is that a "show status like 'tokudb%'" could be nondeterministic in that the variable names could be returned in different ordering. I believe both InnoDB and TokuDB rely on their testing of information_schema.global_status for these where a sorted query against the tables could be performed (which is now supposed to be performance_schema only in 5.7).

**Comment by laurynas-biveinis** [comment]
I'd say this is an InnoDB test hole too. At least PFS way of querying status vars is now tested in 5.7. Are there I_S/PFS tests for status vars? I think none either? SHOW STATUS LIKE can be unrolled to test each show for each variable individually for sorting concerns. I believe something must be tested, just needs to be decided what exactly. The options are 5.6 SHOW STATUS LIKE, I_S.GLOBAL|SESSION_STATUS, the same in 5.7, and 5.7 PFS status tables. IMHO Ideally everything, but 5.7-only PFS could be OK.

**Comment by george-lorch** [comment]
I agree sort of and was surprised that I could not find any InnoDB comparable test. We could/should probably have a test that individually selects the hundreds of status variables by name to ensure they exist, but it would not do much in validating if one was added without making a test change.

The status variables are tested in 5.6 through I_S.GLOBAL. TokuDB has the same test ported (tokudb.information-schema-global-status) with the addition that it absorbs the deprecated warning. I do not know why or how this test passed before this fix.

The test does nothing more than "select variable_name from information_schema.global_status where variable_name like 'tokudb%' order by variable_name asc;"

The result contains the full list of status variables. In 5.7, there is the additional sql warning "Warning 1287  'INFORMATION_SCHEMA.GLOBAL_STATUS' is deprecated and will be removed in a future release. Please use performance_schema.global_status instead"

So I can see changing the test name/function to use the PFS tables, but I think anything beyond that is probably out of scope for this specific issue and should probably have another that is a broader fix to the entire issue of global/local status testing.

**Comment by laurynas-biveinis** [comment]
Check perfschema.show_sanity for how it works, I think it tests that all server variables are present. I agree that this doesn't have to be done in the current PR, just please log it somewhere. Looking at information-schema-global-status current result file, it shows a non-empty list of the status variables. So if the variables are present in I_S but not SHOW STATUS, then a SHOW STATUS test is indeed needed. PFS tests should be added in addition to I_S ones, not replaced, as long as I_S still works even if deprecated.

**Comment by george-lorch** [comment]
I don't disagree on a SHOW STATUS test and I looked to create one when I realized the problem. I just see no existing example from InnoDB nor a way to create a deterministic one that would catch the addition of (but not re-recorded) AND missing status in one shot. Missing can be detected by individually selecting each, but added? Then this also goes to 5.6, if such a test could be devised, it should be originated there properly forward merged to 5.7 AND a similarly critical bug should be introduced against InnoDB/XtraDB for missing the test. It makes me wonder really how/when the ordering of the listing of status variables might change and force a re-record such that a simple "show status like..." would be a problematic solution. I just see how TokuDB reports them by pulling a list from FT which leaves TokuDBs ordering entirely at the mercy of FT.

**Comment by laurynas-biveinis** [comment]
Right. This can be tested with I_S/PFS, but I don't see a good way for SHOW STATUS, but could be missing something. Let's revisit when time comes.

**Comment by george-lorch** [comment]
OK, something interesting. I created a new simple test that is similar to information-schema-global-status.test and recorded it with my code changes in place. I then reverted my code changes and ran both the information-schema-global-status.test and my new performance-schema-global-status.test

The information-schema-global-status.test still passed where the new performance-schema-global-status.test failed. The two result files are nearly identical with the exception of the case of the reported variable names and the warning in the information-schema-global-status.test.

I think adding this test to this PR is enough and then we create a new issue to investigate SHOW STATUS test for both engines.

File: mysql-test/suite/tokudb/t/performance-schema-global-status.test [hunk 1]
```diff
@@ -0,0 +1,4 @@
+source include/have_tokudb.inc;
```
Comment (laurynas-biveinis) [inline mysql-test/suite/tokudb/t/performance-schema-global-status.test:1]: - source include/have_perfschema.inc

**Comment by laurynas-biveinis** [comment]
LGTM with one trivial change

**Pull request merged**